src/Syncs/BloomFilterSync.cpp

```diff
@@ -0,0 +1,217 @@
+
+        // connect to server
+        mySyncStats.timerStart(SyncStats::IDLE_TIME);
+        commSync->commConnect();
+        mySyncStats.timerEnd(SyncStats::IDLE_TIME);
+
+        // Verify server and client Bloom Filters have matching size and numHashes otherwise fail
+        mySyncStats.timerStart(SyncStats::COMM_TIME);
+
+        int clientBFsize = myBloomFilter.getSize();
+        int clientNumHashes = myBloomFilter.getNumHashes();
+
+        int serverBFsize = commSync->commRecv_int();
+        int serverNumHashes = commSync->commRecv_int();
+
+        commSync->commSend(clientBFsize);
+        commSync->commSend(clientNumHashes);
+
+        if((clientBFsize != serverBFsize) || (clientNumHashes != serverNumHashes)) {
+            Logger::gLog(Logger::METHOD_DETAILS, "BloomFilter parameters do not match up between client and server!");
+            mySyncStats.timerEnd(SyncStats::COMM_TIME);
+            mySyncStats.increment(SyncStats::XMIT,commSync->getXmitBytes());
+            mySyncStats.increment(SyncStats::RECV,commSync->getRecvBytes());
+            return false;
+        }
+        mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+        // send client's bloomFilter to server
+        mySyncStats.timerStart(SyncStats::COMP_TIME);
+        ZZ myBFZZ = myBloomFilter.toZZ();
+        mySyncStats.timerEnd(SyncStats::COMP_TIME);
+	mySyncStats.timerStart(SyncStats::COMM_TIME);
+	commSync->commSend(myBFZZ);
+        mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+	// Recieve OMS list from server's computation
+	mySyncStats.timerStart(SyncStats::COMM_TIME);
+        list<shared_ptr<DataObject>> newOMS = commSync->commRecv_DataObject_List();
+        mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+        mySyncStats.timerStart(SyncStats::COMP_TIME);
+        otherMinusSelf.insert(otherMinusSelf.end(), newOMS.begin(), newOMS.end());
+        mySyncStats.timerEnd(SyncStats::COMP_TIME);
+
+        // receive server's bloomFilter
+        mySyncStats.timerStart(SyncStats::COMM_TIME);
+        ZZ theirBFZZ = commSync->commRecv_ZZ();
+        mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+        // Determine SMO list from server's bloom filter
+        mySyncStats.timerStart(SyncStats::COMP_TIME);
+        BloomFilter theirBF = myBloomFilter.ZZtoBF(theirBFZZ);
+	for(auto iter = SyncMethod::beginElements(); iter != SyncMethod::endElements(); iter++)
+	{
+		if(!theirBF.exist((**iter).to_ZZ()))
+                {
+                        selfMinusOther.push_back(make_shared<DataObject>(**iter));
+                }
+	}
+        mySyncStats.timerEnd(SyncStats::COMP_TIME);
+
+        // Send SMO list to server
+        mySyncStats.timerStart(SyncStats::COMM_TIME);
+        commSync->commSend(selfMinusOther);
+        mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+        stringstream msg;
+        msg << "BloomFilterSync succeeded." << endl;
+        msg << "self - other = " << printListOfSharedPtrs(selfMinusOther) << endl;
```

Comment:
This will be calculated regardless of logging state.

Reply:
I implemented a try catch block for both SyncClient and SyncServer so that a failure message will be logged in the case of failure. This is similar to the implementation in the Cuckoo Sync and IBLT Sync.